framework/auth/core.py

```diff
@@ -118,9 +118,16 @@ def get_user(email=None, password=None, verification_key=None):
                              'be provided.')
 
     query_list = []
+    if oauth_provider and oauth_id:
+        # TODO: this query is not working, need ot be fixed
```

Comment:
As discussed, `query_list.append(Q('oauth.{}'.format(oauth_provider), 'eq', oauth_id))` should work, with a slight change to how the `oauth` dict is stored